### ISSUE TITLE
Collision exceptions don't work with TileMap node

### ISSUE THREAD

**Issue by nmearl** [godotengine/godot#17090]
**Godot version:**
v3.0.1.stable.official

**OS/device including version:**
elementary OS 0.4.1 Loki
Ubuntu 16.04.3 LTS
Linux 4.13.0-36-generic

**Issue description:**
Using the `add_collision_exception_with` method of a `KinematicBody2D` when the other `Node` is a `TileMap` does not result in collisions being excepted.

Expected behavior that `KinematicBody2D` falls through `TileMap`.

**Steps to reproduce:**
- Generate `Tileset` from scene with hierarchy
```
+ Node2D
  + Sprite
    + StaticBody2D
      - CollisionPolygon2D
```
- Create new scene with a `TileMap` and `KinematicBody2D`.
- Add `Tileset` to `TileMap` and create a platform for the `KinematicBody2D`.
- On key event, attempt to add collision exception between `TileMap` and `KinematicBody2D` using `add_collision_exception_with(get_node("../TileMap"))`

**Comment by danilo2205** [comment]
I have the same problem :/

**Comment by KoBeWi** [comment]
Still valid in 07db4d57d

Here's MRP:
[ReproductionProject.zip](https://github.com/godotengine/godot/files/4918243/ReproductionProject.zip)

**Comment by jakefriend-dev** [comment]
For anyone coming across this and looking for a mid-term solution for raycasting past a TileMap, I worked around this with a process of replacing TileMap-based exception addds with shifting the Raycast2D along its own length, then shortening the length.

So, instead of adding an exception, the Raycast2D is moved along its length by a set increment (and shortened, so that the initial distance doesn't get extended). You continue to loop as if there's an exception, and eventually you'll either get past the TileMap collision, see something else, or run out of length to increment. Seems to work fine in practice! You do need to remember to reset the raycast's length at the start of your loop.

**Comment by knightofiam** [comment]
Confirming issue still exists in 3.5 stable with `RayCast2D`.

**Comment by knightofiam** [comment]
> For anyone coming across this and looking for a mid-term solution for raycasting past a TileMap, I worked around this with a process of replacing TileMap-based exception addds with shifting the Raycast2D along its own length, then shortening the length.
> 
> So, instead of adding an exception, the Raycast2D is moved along its length by a set increment (and shortened, so that the initial distance doesn't get extended). You continue to loop as if there's an exception, and eventually you'll either get past the TileMap collision, see something else, or run out of length to increment. Seems to work fine in practice! You do need to remember to reset the raycast's length at the start of your loop.

There's a simpler `TileMap` workaround in case you're looping `RayCast2D` collisions multiple times per frame to get all of them instead of just the first one it hits:

1. In the loop, check if the collision object is a `TileMap` and if so add it manually to a custom list of `TileMap` exceptions.

2. Each iteration check if the collision object is both a `TileMap` and in the exception list and if so ignore it. 

3. Force the raycast to update in the same frame using `RayCast2D#ForceRaycastUpdate`.

(Don't forget to clear the exception list first, outside the loop.)

You can see my full working code for this [here](https://github.com/forerunnergames/coa/blob/develop/scripts/Dropdown.cs#L32).

**Comment by akien-mga** [comment]
Closed by #73851.
